src/main/java/com/harriet/shopiify/order/service/OrderService.java

```diff
@@ -0,0 +1,203 @@
+package com.harriet.shopiify.order.service;
+
+import com.harriet.shopiify.auth.repository.ShippingAddressRepository;
+import com.harriet.shopiify.cart.model.CartItem;
+import com.harriet.shopiify.cart.model.CartItemKey;
+import com.harriet.shopiify.cart.repository.CartItemRepository;
+import com.harriet.shopiify.order.dto.OrderDTO;
+import com.harriet.shopiify.order.dto.OrderItemDTO;
+import com.harriet.shopiify.order.model.Order;
+import com.harriet.shopiify.order.model.OrderItem;
+import com.harriet.shopiify.order.model.OrderItemKey;
+import com.harriet.shopiify.order.repository.OrderItemRepository;
+import com.harriet.shopiify.order.repository.OrderRepository;
+import com.harriet.shopiify.order.repository.OrderStatusRepository;
+import com.harriet.shopiify.order.vo.OrderCreateVO;
+import com.harriet.shopiify.order.vo.OrderItemVO;
+import com.harriet.shopiify.product.model.Product;
+import com.harriet.shopiify.product.repository.ProductRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.aspectj.weaver.ast.Or;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class OrderService {
+
+    private final ShippingAddressRepository shippingAddressRepository;
+    private final OrderStatusRepository orderStatusRepository;
+    private final ProductRepository productRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Transactional
+    public Long createOrder(OrderCreateVO vo) throws Exception{
+        Long createdOrderId = null;
+        try{
+            createdOrderId = saveOrder(vo);
+            saveOrderItems(vo);
+            deleteCartItems(vo);
+            reduceStockByOrderedQty(vo);
+        } catch(Exception e){
+            throw new Exception(e.getMessage());
+        }
+        return createdOrderId;
+    }
+
+    private Long saveOrder(OrderCreateVO vo) throws Exception{
+        log.info("creating new order");
+        if (vo.getOrderItems().isEmpty()){
+            throw new UnsupportedOperationException("cannot create new order with no order items");
+        }
+        for ( OrderItemVO item : vo.getOrderItems()){
+            log.info("cart id and product id: {}, {} ",vo.getCartId(), item.getProductId());
+            Optional<CartItem> cartItem = cartItemRepository.findById(toCartItemKey(vo.getCartId(), item.getProductId()));
+            Boolean itemInCart = cartItem.isPresent();
+            log.info("itemInCart: {} ", itemInCart);
+            Boolean orderQtySameInCart = itemInCart ? cartItem.get().getQuantity()==item.getQuantity() : false;
```

Comment:
Can just be `itemInCart && cartItem.get().getQuantity()==item.getQuantity()` instead.

Reply:
noted on this leaner code pattern.
anyway, I have modified the OrderCreateVO to take in a list of CartItem ( cartId and productId), hence this check for quantity match is no longer needed.